scripts/globals/combat/physical_utilities.lua

```diff
@@ -294,7 +325,168 @@ xi.combat.physical.calculateFTP = function(actor, wsTP1000, wsTP2000, wsTP3000)
     return fTP
 end
 
-xi.combat.physical.calculatePDIF = function(actor, additionalParamsHere)
+xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAttackMod, isCritical, applyLevelCorrection, tpIgnoresDefense, tpFactor)
+    local pDif = 0
+
+    -- Step 1: Attack / Defense Ratio
+    local baseRatio     = 0
+    local actorAttack   = math.floor(actor:getStat(xi.mod.ATT) * wsAttackMod)
+    local targetDefense = target:getStat(xi.mod.DEF)
+
+    -- Actor Attack modifiers.
+    if actor:hasStatusEffect(xi.effect.BUILDING_FLOURISH) then
+        local flourishEffect = actor:getStatusEffect(xi.effect.BUILDING_FLOURISH)
+
+        if flourishEffect:getPower() >= 2 then -- 2 or more Finishing Moves used.
+            actorAttack = actorAttack + 25 + flourishEffect:getSubPower()
+        end
+    end
+
+    -- Target Defense Modifiers.
+    local ignoreDefenseFactor = 1
+
+    if tpIgnoresDefense then
+        ignoreDefenseFactor = tpFactor
+    end
+
+    targetDefense = math.floor(targetDefense * ignoreDefenseFactor)
+
+    baseRatio = actorAttack / targetDefense
+
+    -- Step 2: cRatio (Level correction, corrected ratio) Zone based!
+    local levelDifFactor = 0
+
+    if applyLevelCorrection then
+        levelDifFactor = (target:getMainLvl() - actor:getMainLvl()) * 0.05
+    end
+
+    local cRatio = baseRatio - levelDifFactor
```

Comment:
baseRatio (Attack/Defense) should be capped prior to hitting the level correction and cRatio
![image](https://user-images.githubusercontent.com/105882754/224742485-0f620e98-6966-47c8-ab4c-f45c74ace288.png)

Reply:
I added the caps to att/def ratio. TY^^